flowmachine/flowmachine/core/server/action_handlers.py

```diff
@@ -183,7 +188,7 @@ def _get_query_kind_for_query_id(query_id: str) -> Union[None, str]:
             status="error", msg=f"Unknown query id: '{query_id}'", payload=payload
         )
     else:
-        q_state_machine = QueryStateMachine(Query.redis, query_id)
+        q_state_machine = QueryStateMachine(get_redis(), query_id, get_db().conn_id)
```

Comment:
There's an argument to be made here that the config param should be replaced by a contextvar in the action handers.